wireguard/wireguard_freebsd.go

```diff
@@ -1,98 +1,157 @@
+	}
 }
 
-// netLink.Attrs - implements required function of NetLink package
-func (l *netLink) Attrs() *netlink.LinkAttrs {
-	return l.attrs
+// netLink.ApplyAddrs - applies the assigned node addresses to given interface (netLink)
+func (nc *NCIface) ApplyAddrs() error {
+	return nil
 }
 
-// netLink.Type - returns type of link i.e wireguard
-func (l *netLink) Type() string {
-	return "wireguard"
+// Configure does nothing on freebsd, all configuration is done by NCIface.Create()
+func Configure() error {
+	return nil
 }
 
-// NCIface.Close closes netmaker interface
-func (n *NCIface) Close() {
-	link := n.getKernelLink()
-	link.Close()
+// SetPeers - sets peers on netmaker WireGuard interface
+func SetPeers() error {
+	nc := GetInterface()
+	nc.Config.Peers = []wgtypes.PeerConfig{}
+	for _, node := range config.GetNodes() {
+		for _, peer := range node.Peers {
+			nc.Config.Peers = append(nc.Config.Peers, peer)
+			cmd := fmt.Sprintf("wg set %s peer %s endpoint %s ", nc.Name, peer.PublicKey, peer.Endpoint)
+			for _, ip := range peer.AllowedIPs {
+				cmd = cmd + "allowed-ips " + ip.String() + " "
+			}
+			if _, err := ncutils.RunCmd(cmd, true); err != nil {
+				return fmt.Errorf("error adding peers %w", err)
+			}
+		}
+	}
+	return nil
 }
 
-// netLink.Close - required function to close linux interface
-func (l *netLink) Close() error {
-	return netlink.LinkDel(l)
-}
+// == private ==
 
-// netLink.ApplyAddrs - applies the assigned node addresses to given interface (netLink)
-func (nc *NCIface) ApplyAddrs() error {
-	l := nc.getKernelLink()
-	currentAddrs, err := netlink.AddrList(l, 0)
+func create(nc *NCIface) error {
+	ifconfig, err := exec.LookPath("ifconfig")
 	if err != nil {
 		return err
 	}
-	if len(currentAddrs) > 0 {
-		for i := range currentAddrs {
-			err = netlink.AddrDel(l, &currentAddrs[i])
-			if err != nil {
-				return err
-			}
+	wg, err := exec.LookPath("wg")
+	if err != nil {
+		return err
+	}
+	if _, err := ncutils.RunCmd(ifconfig+" "+nc.Name, true); err == nil {
+		if _, err := ncutils.RunCmd(ifconfig+" "+nc.Name+" destroy", true); err != nil {
+			return err
 		}
 	}
-	for _, node := range config.GetNodes() {
-		var address netlink.Addr
-		var address6 netlink.Addr
-		address.IPNet = &node.Address
-		if address.IPNet.IP != nil {
-			logger.Log(3, "adding address ", address.IP.String(), "to netmaker address")
-			if err := netlink.AddrAdd(l, &address); err != nil {
-				logger.Log(0, "error adding addr", err.Error())
-				return err
+	if _, err := ncutils.RunCmd(ifconfig+" wg create name "+nc.Name, true); err != nil {
+		return err
+	}
+	//config
+	if err := strip(); err != nil {
+		return err
+	}
+	if err != nil {
+		return err
+	}
+	if _, err := ncutils.RunCmd(fmt.Sprintf("%s setconf %s %s", wg, nc.Name, os.TempDir()+"/netmaker.conf"), true); err != nil {
+		return err
+	}
+	if err := os.Remove(os.TempDir() + "/netmaker.conf"); err != nil {
+		return err
+	}
+	//add addresses
+	for _, address := range nc.Addresses {
```

Comment:
is there a reason this logic isn't in the ApplyAddrs func?

Reply:
AppyAddrs is never called in freebsd.